packages/react-storage/src/components/StorageBrowser/actions/handlers/utils.ts

```diff
@@ -18,7 +19,34 @@ export const constructBucket = ({
   region: string;
 } => ({ bucketName, region });
 
-export const parseLocationAccess = (location: LocationAccess): LocationData => {
+export const parseAccessGrantLocationScope = (
+  scope: string,
+  type: LocationType
+): { bucket: string; prefix: string } => {
+  const slicedScope = scope.slice(5);
+  if (type === 'BUCKET') {
+    // { scope: 's3://bucket/*', type: 'BUCKET', },
+    const bucket = slicedScope.slice(0, -2);
+    const prefix = '';
+    return { bucket, prefix };
+  } else if (type === 'PREFIX') {
+    // { scope: 's3://bucket/path/*', type: 'PREFIX', },
+    const bucket = slicedScope.slice(0, slicedScope.indexOf('/'));
+    const prefix = `${slicedScope.slice(bucket.length + 1, -1)}`;
+    return { bucket, prefix };
+  } else if (type === 'OBJECT') {
+    // { scope: 's3://bucket/path/to/object', type: 'OBJECT', },
+    const bucket = slicedScope.slice(0, slicedScope.indexOf('/'));
+    const prefix = slicedScope.slice(bucket.length + 1);
+    return { bucket, prefix };
+  } else {
+    throw new Error(`Invalid location type: ${type}`);
+  }
+};
+
+export const parseAccessGrantLocation = (
```

Comment:
Naming is confusing here - `parseAccessGrantLocation` and there is `parseAccessGrantLocations`

Reply:
This is only used in a deprecating code path in `do-not-import` folder. It should be removed very soon.

Reply:
yes got it as going through down below. Makes sense. lets make sure to remove this and not forget it

Reply:
Should we add a TODO or FIXME similar to what u have already